packages/interaction-tracking/src/InteractionTrackingSubscriptions.js

```diff
@@ -0,0 +1,161 @@
+if (enableInteractionTracking) {
+  subscribers = new Set();
+}
+
+export function subscribe(subscriber: Subscriber): void {
+  subscribers.add(subscriber);
+}
+
+export function unsubscribe(subscriber: Subscriber): void {
+  subscribers.delete(subscriber);
+}
+
+function onInteractionTracked(interaction: Interaction): void {
+  let didCatchError = false;
+  let caughtError = null;
+
+  subscribers.forEach(subscriber => {
+    try {
+      subscriber.onInteractionTracked(interaction);
+    } catch (error) {
+      if (!didCatchError) {
+        didCatchError = true;
+        caughtError = error;
+      }
+    }
+  });
+
+  if (didCatchError) {
+    throw caughtError;
+  }
+}
+
+function onInteractionScheduledWorkCompleted(interaction: Interaction): void {
+  let didCatchError = false;
+  let caughtError = null;
+
+  subscribers.forEach(subscriber => {
+    try {
+      subscriber.onInteractionScheduledWorkCompleted(interaction);
+    } catch (error) {
+      if (!didCatchError) {
+        didCatchError = true;
+        caughtError = error;
+      }
+    }
+  });
+
+  if (didCatchError) {
+    throw caughtError;
+  }
+}
+
+function onWorkScheduled(
+  interactions: Set<Interaction>,
+  threadID: number,
+): void {
+  let didCatchError = false;
+  let caughtError = null;
+
+  subscribers.forEach(subscriber => {
+    try {
+      subscriber.onWorkScheduled(interactions, threadID);
+    } catch (error) {
+      if (!didCatchError) {
+        didCatchError = true;
+        caughtError = error;
+      }
+    }
+  });
+
+  if (didCatchError) {
+    throw caughtError;
+  }
+}
+
+function onWorkStarted(interactions: Set<Interaction>, threadID: number): void {
+  let didCatchError = false;
+  let caughtError = null;
+
+  subscribers.forEach(subscriber => {
+    try {
+      subscriber.onWorkStarted(interactions, threadID);
+    } catch (error) {
+      if (!didCatchError) {
+        didCatchError = true;
+        caughtError = error;
+      }
+    }
+  });
+
+  if (didCatchError) {
+    throw caughtError;
+  }
+}
+
+function onWorkStopped(interactions: Set<Interaction>, threadID: number): void {
+  let didCatchError = false;
+  let caughtError = null;
+
+  subscribers.forEach(subscriber => {
+    try {
+      subscriber.onWorkStopped(interactions, threadID);
+    } catch (error) {
+      if (!didCatchError) {
+        didCatchError = true;
+        caughtError = error;
+      }
+    }
+  });
+
+  if (didCatchError) {
+    throw caughtError;
+  }
+}
+
+function onWorkCanceled(
+  interactions: Set<Interaction>,
+  threadID: number,
+): void {
+  let didCatchError = false;
+  let caughtError = null;
+
+  subscribers.forEach(subscriber => {
+    try {
+      subscriber.onWorkCanceled(interactions, threadID);
+    } catch (error) {
+      if (!didCatchError) {
+        didCatchError = true;
+        caughtError = error;
+      }
+    }
+  });
+
+  if (didCatchError) {
+    throw caughtError;
+  }
+}
+
+__subscriberRef.current = {
```

Comment:
This shouldn't bother subscribing if `enableInteractionTracking` is false, right?